examples/units/units_scatter.py

```diff
@@ -9,28 +9,25 @@
 arrays.
 """
 import numpy as np
+import matplotlib.pyplot as plt
 from basic_units import secs, hertz, minutes
-from matplotlib.pylab import figure, show
 
 # create masked array
+data = (1, 2, 3, 4, 5, 6, 7, 8)
+mask = (1, 0, 1, 0, 0, 0, 1, 0)
+xsecs = secs * np.ma.MaskedArray(data, mask, float)
 
-
-xsecs = secs*np.ma.MaskedArray((1, 2, 3, 4, 5, 6, 7, 8), (1, 0, 1, 0, 0, 0, 1, 0), float)
-#xsecs = secs*np.arange(1,10.)
-
-fig = figure()
-ax1 = fig.add_subplot(3, 1, 1)
+fig, (ax1, ax2, ax3) = plt.subplots(nrows=3, sharex=True)
 ax1.scatter(xsecs, xsecs)
-#ax1.set_ylabel('seconds')
+ax1.yaxis.set_units(secs)
 ax1.axis([0, 10, 0, 10])
 
-ax2 = fig.add_subplot(3, 1, 2, sharex=ax1)
 ax2.scatter(xsecs, xsecs, yunits=hertz)
 ax2.axis([0, 10, 0, 1])
 
-ax3 = fig.add_subplot(3, 1, 3, sharex=ax1)
 ax3.scatter(xsecs, xsecs, yunits=hertz)
 ax3.yaxis.set_units(minutes)
 ax3.axis([0, 10, 0, 1])
 
-show()
+fig.tight_layout()
+plt.show()
```

Comment:
mine too, but some examples have more than 1 figure, so `pyplot.show()` shows them all.

Happy to change this one though.